wandb/sdk/launch/agent/agent.py

```diff
@@ -121,6 +136,11 @@ def run_job(self, job: Dict[str, Any]) -> None:
         """Sets up project and runs the job."""
         # TODO: logger
         print("agent: got job", job)
+        # update agent status
+        update_ret = self._api.update_launch_agent_status(self._id, AGENT_RUNNING)
```

Comment:
We should probably run the launch process command in another process so the agent can continue to update its status while the job is running. That way, when we add something to check if an agent died without reporting itself killed, we won't mark agents as killed when they've been running long jobs.

Reply:
do we need to do that in this pr? i assumed that would come in the next chunk of work for heartbeats

Reply:
Oh, I thought we'd put heartbeats in this PR. That sounds fine to get the new UI out, but we should prioritize heartbeats after.